test/loader/serializeInstructionTest.cpp

```diff
@@ -666,4 +666,119 @@ TEST(SerializeInstructionTest, SerializeConstInstruction) {
   };
   EXPECT_EQ(Output, Expected);
 }
+
+TEST(SerializeInstructionTest, SerializeFuncRefControlInstruction) {
+  std::vector<uint8_t> Expected;
+  std::vector<uint8_t> Output;
+  std::vector<WasmEdge::AST::Instruction> Instructions;
+
+  // 12. Test typed function references control flow instructions.
+  //
+  //   1.  Serialize br_on_null instruction.
+  //   2.  Serialize br_on_non_null instruction.
```

Comment:
In addition, you can add the fail case (`EXPECT_FALSE`) for testing these instructions without the typed function reference proposal config.